src/d3a/models/market/__init__.py

```diff
@@ -73,6 +230,9 @@ def __init__(self, time_slot=None, bc=None, notification_listener=None, readonly
             self.notification_listeners.append(notification_listener)
         self.current_tick = 0
         self.device_registry = DeviceRegistry.REGISTRY
+        if ConstSettings.GeneralSettings.EVENT_DISPATCHING_VIA_REDIS:
+            self.redis_api = MarketRedisApi(self)
```

Comment:
I couldn't the usage of __self.redis_api__ instance variable. May be I have missed something. .

Reply:
You are right, it is not used anywhere in the code. It needs to be there though for the subscription. In order for the subscribe thread to keep on running, there has to be a reference to this class, because if all references of this object get lost, the object gets destroyed, and the subscribe thread terminates accordingly. Even though run_in_thread(daemon=True) does its best to avoid this situation, it is not guaranteed that it will not get deleted when it goes out of scope, therefore it is good for this to stay as a class member.

Reply:
Thanks for the explanation. Makes sense to me now.